src/BuildScriptGenerator/Node/NodeBashBuildSnippet.sh.tpl

```diff
@@ -15,11 +15,12 @@ echo
 # Yarn config is per user, and since the build might run with a non-root account, we make sure
 # the yarn cache is set on every build.
 YARN_CACHE_DIR=/usr/local/share/yarn-cache
+yarnCacheFolderName={{ YarnCacheFolderName }}
 if [ -d $YARN_CACHE_DIR ]
 then
 	echo
     echo "Configuring Yarn cache folder..."
-    yarn config set cache-folder $YARN_CACHE_DIR
+    yarn config set $yarnCacheFolderName $YARN_CACHE_DIR
```

Comment:
why is this change? are we passing the cache folder name to the script now?

Reply:
This is simply the CLI command difference between yarn 1 and yarn 2, so we need to detect and pass in different command parameter here.